### PR TITLE
Add support for matching on s3 bucket lifecycle rules

### PR DESCRIPTION
**Pull request by azriel91**
Addresses issue #218

### CODE REVIEW

**Comment by k1LoW** [comment]
Hi @azriel91 .
Thank you for your nice commit!!

File: doc/resource_types.md [hunk 1]
```diff
@@ -1613,6 +1613,21 @@ end
 ```
 
 
+### have_lifecycle_rule
```
Comment (k1LoW) [inline doc/resource_types.md]: Append doc to `doc/_resource_types/s3_bucket.md` and run `bundle exec bin/toolbox docgen > doc/resource_types.md` 
( see https://github.com/k1LoW/awspec/blob/master/doc/contributing.md )

Comment (azriel91) [inline doc/resource_types.md]: Thanks! I rushed it a little too early and forgot to run the spec tests

**Comment by k1LoW** [comment]
Thank you 🎊 
LGTM!

**Comment by k1LoW** [comment]
Released as v0.65.0.

**Pull request merged**